From 91c4074b0eb9e4f56e3c182e873f16bdbd83c11e Mon Sep 17 00:00:00 2001
From: h-vetinari <h.vetinari@gmx.com>
Date: Sun, 30 Jun 2024 07:33:27 +1100
Subject: [PATCH 1/2] MAINT: fix typo in small_dynamic_array.h (#21069)

on clang-19, this causes:
```
../scipy/_lib/_uarray/small_dynamic_array.h(145,18): error: reference to non-static member function must be called
  145 |     size_ = copy.size;
      |             ~~~~~^~~~
1 error generated.
```
I'm not sure how previous versions (much less other compilers) dealt with this,
as it seems that the `SmallDynamicArray` class has no `size` member or field at all.
---
 scipy/_lib/_uarray/small_dynamic_array.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scipy/_lib/_uarray/small_dynamic_array.h b/scipy/_lib/_uarray/small_dynamic_array.h
index b6c46d7c44..351b5d8fc6 100644
--- a/scipy/_lib/_uarray/small_dynamic_array.h
+++ b/scipy/_lib/_uarray/small_dynamic_array.h
@@ -142,7 +142,7 @@ public:
 
     clear();
 
-    size_ = copy.size;
+    size_ = copy.size_;
     try {
       allocate();
     } catch (...) {
